### PR TITLE
Add test that sends an invalid userInfo to AppCheck

### PR DESCRIPTION
**Pull request by schmidt-sebastian**

### CODE REVIEW

File: Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm [hunk 1]
```diff
@@ -220,6 +220,45 @@ - (nonnull NSString*)tokenDidChangeNotificationName {
...
+      [token_promise](const std::string& result) {
+        if (result != "") {
+          token_promise->set_value(result);
+        }
+      });
+
+  // Sending this notifcation would cause a crash if it was processed in the
+  // AppCheck notification handlder since AppCheck expects the userInfo object
+  // to an NSDictionary.
+  id userInfo = @"this_should_be_a_dictionary";
```
Comment (maksymmalyhin) [inline Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:242]: Hm, [userInfo](https://developer.apple.com/documentation/foundation/nsnotification/1409222-userinfo?language=objc) is expected to be a dictionary. If it is not then there is a bug in somewhere else on the notification poster side.

Comment (maksymmalyhin) [reply Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:242]: I guess if we expect `userInfo` to be not a dictionary then we should handle it gracefully in the code. From the other side I'm not sure if we should code defensively against such types of issues. I don't feel strongly about it but this specific test seems to be a bit redundant to me.

**Review by maksymmalyhin — APPROVED** [review]
LGTM in general.

File: Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm (hunk 1 above)
Comment (schmidt-sebastian) [reply Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:242]: It's the only way I could reproduce the issue, so while it is a bit redundant it is IMHO worth keeping as it tests the entire problematic code flow, even if the fix is of much narrower scope.

**Pull request merged**